### PR TITLE
Brush up CI, add Windows build

### PR DESCRIPTION
**Pull request by sanmai**
This PR:

  - [x] Adds Windows to the build matrix (so you don't have to install PHP on AppVeyor)
  - [x] Sets different names for each build, so that protected branch can require specific statuses
  - [x] Add Composer cache
  - [x] Removed redundant double quotes
  - [x] Changes steps to multi-line format so that adding a line to a step lead to one line diff

### CODE REVIEW

**Comment by pimjansen** [comment]
Do even need to worry for PHP?
https://news-web.php.net/php.internals/110907

**Comment by sanmai** [comment]
You mean Windows? IDK, it's a single line to remove later.

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -4,36 +4,49 @@ on:
...
-        php-version:
-          - "7.1"
-          - "7.2"
-          - "7.3"
-          - "7.4"
-          - "8.0"
+        operating-system: ['ubuntu-latest']
+        php-version: ['7.1', '7.2', '7.3', '7.4', '8.0']
+        include:
+          - { operating-system: 'windows-latest', php-version: '7.1' }
```
Comment (localheinz) [inline .github/workflows/continuous-integration.yml:18]:
```suggestion
          - operating-system: 'windows-latest'
            php-version: '7.1'
```

**Review by localheinz — COMMENTED** [review]
How do you feel about applying these changes, @sanmai?

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -4,36 +4,49 @@ on:
...
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: composer-${{ runner.os }}-${{ matrix.php-version }}-${{ hashFiles('composer.*') }}
+          restore-keys: |
+            composer-${{ runner.os }}-${{ matrix.php-version }}-
+            composer-${{ runner.os }}-
+            composer-
+
+      - name: Install dependencies with Composer
+        run: |
+          composer update --no-interaction --no-progress
```
[the remark this answers is not in the document]
Comment (sanmai) [reply .github/workflows/continuous-integration.yml:55]: Not a big fan of this format you're suggesting:
- Multi-line format allows adding another command to a step with one line diff.
- It helps readability where, essentially, shell commands are not prefixed with YAML keys. You can clearly see what is a CI command, and where YAML ends and starts, and so on.

**Review by localheinz — APPROVED** [review]

**Comment by sanmai** [comment]
@localheinz Sure. I applied most of the changes, but I'd argue it is more sensible to keep the two bits above.

**Comment by localheinz** [comment]
Thank you, @pimjansen and @sanmai!

**Comment by sanmai** [comment]
Thank you for the review @localheinz

**Pull request merged**